apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteCanonicalTokens.ts

```diff
@@ -65,37 +69,39 @@ export default function useRegisterRemoteCanonicalTokens(
   });
 
   const multicallArgs = useMemo(() => {
-    if (
-      !isValidEVMAddress(input.tokenAddress) || // This is only used for EVM chains
-      !tokenDetails ||
-      !gasFeesData ||
-      tokenDetails.kind !== "canonical" ||
-      chainId === SUI_CHAIN_ID
-    )
-      return [];
-
-    return destinationChainIds.map((axelarChainId, i) => {
-      const gasValue = gasFeesData.gasFees[i].fee;
-
-      return INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteCanonicalInterchainToken.data(
+    if (!isValidEVMAddress(input.tokenAddress)) return [] as `0x${string}`[];
+    if (!tokenDetails) return [] as `0x${string}`[];
+    if (!gasFeesData) return [] as `0x${string}`[];
+    if (tokenDetails.kind !== "canonical") return [] as `0x${string}`[];
+    if (chainId === SUI_CHAIN_ID) return [] as `0x${string}`[];
+
+    const hasLengthMatch =
+      gasFeesData.gasFees.length === destinationChainNames.length;
+    if (!hasLengthMatch) {
+      throw new Error("Gas fees length does not match destination chains");
+    }
+
+    return destinationChainNames.map((destinationChain, i) =>
+      INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteCanonicalInterchainToken.data(
         {
           originalTokenAddress: tokenDetails.tokenAddress as `0x${string}`,
-          destinationChain: axelarChainId,
-          gasValue,
+          destinationChain,
+          gasValue: scaleGasValue(chainId, gasFeesData.gasFees[i].fee),
         }
-      );
-    });
-  }, [tokenDetails, gasFeesData, chainId, input, destinationChainIds]);
+      )
+    );
+  }, [tokenDetails, gasFeesData, chainId, input, destinationChainNames]);
 
   const totalGasFee = gasFeesData?.totalGasFee ?? 0n;
 
-  const { data: config } = useSimulateInterchainTokenFactoryMulticall({
-    value: totalGasFee,
-    args: [multicallArgs],
-    query: {
-      enabled: multicallArgs.length > 0 && totalGasFee > 0n,
-    },
-  });
+  const { data: config, error: simulationError } =
+    useSimulateInterchainTokenFactoryMulticall({
+      value: totalGasFee,
+      args: [multicallArgs],
+      query: {
+        enabled: chainId !== SUI_CHAIN_ID && multicallArgs.length > 0,
```

Comment:
We can create an array for chains that have disabled simulation in order to avoid having chain-specific handling in the body.

This applies to `useRegisterRemoteInterchainTokens.ts`.

Feel free to add those to the chain-specific improvements task if you want us to pick it up in the future.